cmd/libs/go2idl/client-gen/main.go

```diff
@@ -39,21 +40,33 @@ var (
 	fakeClient    = flag.Bool("fake-clientset", true, "when set, client-gen will generate the fake clientset that can be used in tests")
 )
 
-func versionToPath(group string, version string) (path string) {
+func versionToPath(gvPath string, group string, version string) (path string) {
 	const base = "k8s.io/kubernetes/pkg"
 	// special case for the core group
 	if group == "api" {
 		path = filepath.Join(base, "api", version)
 	} else {
-		path = filepath.Join(base, "apis", group, version)
+		path = filepath.Join(base, "apis", gvPath, group, version)
 	}
 	return
 }
 
+func parsePathGroupVersion(pgvString string) (gvPath string, gvString string) {
```

Comment:
yes, it gets the base directory from --input parameter (e.g. --input="../../federation/apis/federation/" will extract "../../federation/apis" and append it upon "k8s.io/kubernetes/pkg/apis" which will resolve "k8s.io/kubernetes/federation/apis"

do you mean that using `--base="federation/apis/" --input="federation/"` and `--base=pkg/apis (default value) --input=extensions/` for k8s?